core/domain/html_cleaner.py

```diff
@@ -400,7 +452,40 @@ def convert_to_text_angular(html_data):
     # html strings they are stored as <br>. Since both of these
     # should match and <br> and <br/> have same working,
     # so the tag has to be replaced in this way.
-    return str(soup).replace('<br/>', '<br>')
+    return unicode(soup).replace('<br/>', '<br>')
+
+
+def convert_tag_contents_to_text_angular(html_data):
+    """This functions converts content with tabs and collapsible
+    components to textangular format.
+
+    Args:
+        html_data: str. The HTML string whose content is to be converted.
+
+    Returns:
+        str. The HTML string with converted content within tag.
+    """
+    soup = bs4.BeautifulSoup(html_data.encode('utf-8'), 'html.parser')
+
+    for collapsible in soup.findAll('oppia-noninteractive-collapsible'):
+        content_html = unescape_html(collapsible['content-with-value'])
+        # This is done because the quotes are a part of content at position 0
+        # and -1. When content is converted to textangular, the quotes get
+        # wrapped within a p tag. To avoid this these first and last quotes
+        # removed before conversion to Textangular and then added back.
+        collapsible['content-with-value'] = '&quot;' + escape_html(
+            convert_to_text_angular(content_html[1:-1])) + '&quot;'
```

Comment:
No, there should be no string concatenation here. Convert to JSON / from JSON as needed, following exactly what happens in the frontend. If the frontend doesn't do content_html[1:-1] then you should not do it here either.

Reply:
Fixed!